### ISSUE TITLE
Idea: Improved User Management

### ISSUE THREAD

**Issue by felixstuermer** [kanru/wp-cf-access-jwt-assertion-login#3]
**I would like to suggest an idea on how to improve the user-workflow within Wordpress vs. allowed users within Cloudflare:**

Currently if a user is not available in Wordpress the whole Login via the Plugin / Cloudflare does not work and this message is stated: _User not available_

What about an improvement of this process with the creation of a new user if the user is theoretically allowed by cloudflare to login but the user in Wordpress is not yet available? 

The master-admin of the website would be able to define a default user-role for everyone that will be created in that way or assign the role each time individually and receives a message with an approval process? This would be pretty cool! 

Really like your plugin! 

All the best from Germany,
Felix

**Comment by kanru** [comment]
Indeed, that sounds a nice enhancement. No promises if it will be implemented, but PR is always welcomed.

Notes for future implementer
* It looks like we can use [wp_create_user](https://developer.wordpress.org/reference/functions/wp_create_user/) to create new user with site default role
* Use [wp_insert_user](https://developer.wordpress.org/reference/functions/wp_insert_user/) for more detailed control
